verifiable-db/src/query/batching/row_chunk/consecutive_rows.rs

```diff
@@ -0,0 +1,1290 @@
+    // this always if we need to "strictly" prove that 2 nodes are consecutive, which happens in the following cases:
+    // - if nodes are in the index tree
+    // - if nodes are in a rows tree, but `first_node_succ_in_range` is true. Indeed, if the successor of first node
+    //   is out of range or doesn't exist, then it means that second node belongs to another rows tree, and so it cannot
+    //   be a successor of first node in the same rows tree
+    let either_is_in_path = b.or(
+        first.successor_info.is_in_path,
+        second.predecessor_info.is_in_path,
+    );
+
+    if !are_rows_tree_nodes {
+        // in case of index tree, we need to enforce that `either_is_in_path` must be true
+        are_consecutive = b.and(are_consecutive, either_is_in_path);
+        // furthermore, we also need to enforce that first_node_succ_in_range and second_node_pred_in_range
+        // are both true; otherwise, the prover could provide the nodes at the boundary and prove them
+        // to be consecutive, which is not ok in the index tree
+        are_consecutive = b.and(are_consecutive, first_node_succ_in_range);
+        are_consecutive = b.and(are_consecutive, second_node_pred_in_range);
+    } else {
+        // in case of rows tree nodes, we need to check that `first_row_succ_in_range == second_row_pred_in_range`,
+        // which should always hold for consecutive rows since:
+        // - if the successor of first row is in range, then second row must be its successor
+        //   in the same rows tree, and so the predecessor of second row is the first row itself,
+        //   which is expected to be in range since we never need to prove nodes not in range
+        //   but with a successor in range
+        // - if the successor of first row is out of range, then second row is expected to
+        //   be a node in the "subsequent" rows tree (i.e., the rows tree stored in the index
+        //   tree node which is the successor of the index tree node storing first row); this node
+        //   can be either:
+        //	    - the first node of the "subsequent" rows tree with value >= min_secondary;
+        //        in this case, the predecessor of second row is < min_secondary, and so out of range
+        //	    - if no such node can be found in the "subsequent" rows tree, then second row will be
+        //        the last node in the "subsequent" rows tree with value < MIN_secondary; in
+        //	      this case, also its predecessor will necessarily be < MIN_secondary, and so
+        //	      out of range
+        // we first compute first_row_succ_in_range XOR second_row_pred_in_range: a XOR b = a + b - 2*a*b
+        let range_flags_sum = b.add(
+            first_node_succ_in_range.target,
+            second_node_pred_in_range.target,
+        );
+        let range_flags_xor = b.arithmetic(
+            F::NEG_ONE + F::NEG_ONE,
```

Comment:
worth putting it in a constant ?

Reply:
Done in commit https://github.com/Lagrange-Labs/mapreduce-plonky2/pull/396/commits/5787dc75a5d6df1a467910e6fc2b524e21943460